compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12922,6 +13065,7 @@ TR::Node *ificmpeqSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier
 
 
    addressCompareConversion(node, s);
+   removeArithmeticsUnderIntegralCompare(node, s);
```

Comment:
It looks like `addressCompareConversion()` can transmute `node` into an address comparison. At the moment `removeArithmeticsUnderIntegralCompare()` should ignore address comparisons due to its explicit list of accepted left child opcodes. But if you want to use `isAdd()`, then just be aware that `aladd` and `aiadd` respond with true

Reply:
With the new assertion at the beginning of `removeArithmeticsUnderIntegralCompare()`, this should guard against passing in `node` when it has been transmuted into an address comparison